### PR TITLE
fix(core,test): fix integration test

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

This PR is aimed at fixing some randomly failed integration tests in ci. 

1. Socket hangup
e.g.
image

Remove the integration test API timeout limit. In case any late response occurs.

2. To improve the integration tests API performance, remove all the external API calls in our test cases. (Should only depend on the local logto service)

e.g.
image

- Remove the Google IdP issuer dependency and replace it with Logto OIDC
- Remove the SAML Azure metadata URL dependency.  Use locally static XML metadata file upload instead.
- Remove the Google, OKTA, and Azure SSO connector creation test cases, as the flow is all the same. Only keep OIDC and SAML. 

4. Fix the API status
e.g. 
image
Post SSO connectors API should throw a 400 error status if the input connector config validation fails.  Add the status code 400 to the koaGuard. 

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/integration-tests/src/tests/console/sso-connectors/sso-connectors-test-cases.ts
Comment (gao-sun) [inline packages/integration-tests/src/tests/console/sso-connectors/sso-connectors-test-cases.ts:1]: I believe there's another remote api call: password pwned test. is it possible to mock a local client for that? no hurries though

**Review by wangsijie — APPROVED** [review]

File: packages/integration-tests/src/tests/console/sso-connectors/helpers.ts [hunk 1]
```diff
@@ -128,6 +132,58 @@ const configureSsoConnectorConnection = async (
...
+      const uploadMetadataXmlOption = Array.from(optionElement).find(
+        (element) => element.textContent === 'Upload the metadata XML file'
+      );
+
+      if (uploadMetadataXmlOption) {
+        const clickEvent = new MouseEvent('click', { bubbles: true });
+        uploadMetadataXmlOption.dispatchEvent(clickEvent);
+      }
+    }
+  }, dropdownMenu);
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/integration-tests/src/tests/console/sso-connectors/helpers.ts:165]: puppeteer page.click API causes unexpected behavior under the headless mode when the element does not contain a native click event, e.g. div element.  That is why we need to dispatch a click event here.

[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/integration-tests/src/tests/console/sso-connectors/helpers.ts:165]: Also, that is why we had multiple waitForTimeout implemented.

[the remark this answers is not in the document]
Comment (charIeszhao) [reply packages/integration-tests/src/tests/console/sso-connectors/helpers.ts:165]: Interesting... Then probably we should also consider changing the implementation in console. Anyways, that should be a non-blocker thing for now

**Review by charIeszhao — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

**Pull request merged**